packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/VXZoom/VXZoom.js

```diff
@@ -0,0 +1,160 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { localPoint } from '@vx/event'
+import { Zoom } from '@vx/zoom'
+import ZoomEventLayer from '../ZoomEventLayer'
+
+class VXZoom extends PureComponent {
+  constructor(props) {
+    super(props)
+    const { setOnZoom } = props
+
+    setOnZoom(this.handleToolbarZoom.bind(this))
+
+    this.onDoubleClick = this.onDoubleClick.bind(this)
+    this.onMouseLeave = this.onMouseLeave.bind(this)
+  }
+
+  static zoom = null
+
+  handleToolbarZoom(type) {
+    const doZoom = {
+      'zoomin': this.zoomIn.bind(this),
+      'zoomout': this.zoomOut.bind(this),
+      'zoomto': this.zoomReset.bind(this)
+    }
+
+    if (doZoom[type]) {
+      doZoom[type]()
+    }
+  }
+
+  zoomIn() {
+    if (!this.props.zooming) return
+    const { zoomInValue } = this.props.zoomConfiguration
+    this.zoom.scale({ scaleX: zoomInValue, scaleY: zoomInValue })
+  }
+
+  zoomOut() {
+    if (!this.props.zooming) return
+    const { zoomOutValue } = this.props.zoomConfiguration
+    this.zoom.scale({ scaleX: zoomOutValue, scaleY: zoomOutValue })
+  }
+
+  zoomReset() {
+    if (!this.props.zooming) return
+    this.zoom.reset()
+  }
+
+  zoomToPoint(event, zoomDirection) {
+    const { zoomInValue, zoomOutValue } = this.props.zoomConfiguration
+    const zoomValue = (zoomDirection === 'in') ? zoomInValue : zoomOutValue
+    const point = localPoint(event)
+    this.zoom.scale({ scaleX: zoomValue, scaleY: zoomValue, point })
+  }
+
+  onDoubleClick(event) {
+    if (this.props.zooming) {
+      this.zoomToPoint(event, 'in')
+    } else {
+      event.preventDefault()
+    }
+  }
+
+  onMouseLeave() {
+    if (!this.zoom.isDragging && !this.props.panning) return
+    this.zoom.dragEnd()
+  }
+
+  onWheel(event) {
+    // performance of this is pretty bad
+    if (this.props.zooming) {
+      const zoomDirection = (-event.deltaY > 0) ? 'in' : 'out'
+      this.zoomToPoint(event, zoomDirection)
+    } else {
+      event.preventDefault()
```

Comment:
Query: I'm not sure what's the purpose of `event.preventDefault()` here? But it may not matter because...

Quirk: ...it seems that it's never triggered? If `!props.zooming` (i.e. by disabling the 'zooming' property in the Storybook knob for ScatterPlotViewer), it seems that `onWheel()` is never called.

Reply:
OK, I see where the quirk comes in: in `ScatterPlotViewer.js`, if that component's `zooming` property is set to false, it renders `<ScatterPlot>` instead of `<ZoomingScatterPlot>`, meaning the `<VXZoom>` rect isn't rendered at all.

Reply:
Yes, that's true with how I've implemented it with the scatter plot, but in theory the `VXZoom` component can be potentially integrated elsewhere for other subject viewers.... This might be cruft, but I'm covering my bases here I think.